src/components/Upload.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+
+const Upload = ({ setCurrentScript, titles, setTitles }) => {
+  const MAXFILESIZE = 50 * 1024 * 1024; // 50MB
+  const [file, setFile] = useState();
+  const [errMessage, setErrMessage] = useState({ message: '', display: 'none', color: 'red' });
+
+  const handleFileChange = (e) => {
+    e.preventDefault();
+    // check that the file size is less than 50 before allowing the upload
+    if (e.target.files) {
+      if (e.target.files[0].size > MAXFILESIZE) {
+        setErrMessage({ message: 'Selected file is too large. Max size is 50MB', display: 'block', color: 'red' });
+        setFile(null);
+      } else {
+        setFile(e.target.files[0]);
+        setErrMessage({ message: '', display: 'none' });
+      }
+    }
+  };
+
+  const handleUploadClick = () => {
+    if (!file) {
+      return;
+    }
+
+    // set the file as Form Data and send it to the server
+    const formData = new FormData();
+    formData.append('newScript', file);
+
+    fetch('/script', {
+      method: 'POST',
+      body: formData,
+    })
+      .then((response) => {
+        if (response.status === 409) {
+          throw new Error('script already exists');
+        }
+        return response.json();
+      })
+      .then((title) => {
+        if (title) {
```

Comment:
Is it possible for title to not be true here? If it isn't, we could probably get rid of the `if`. If it can be false, then shouldn't we handle that case?

Reply:
Good catch. I had been using this to make sure I wasn't setting a title when we got a 409 response that the script was already in the database. When I realized I should be throwing an error in that case, that made this 'if' redundant.